tests/elemental3/generate_image.pm

```diff
@@ -92,23 +96,33 @@ sub run {
 
     # Get the system extensions
     foreach my $sysext (@sysexts) {
-        assert_script_run("curl -f ${sysext_path}/${sysext} -o ${sysext_dir}/${sysext}", 300);
+        assert_script_run(
+            "curl -v -o ${sysext_dir}/${sysext} ${sysext_path}/${sysext}",
+            300);
     }
 
     # Package the system extensions
     assert_script_run("tar cvaf $overlay -C $sysext_root .");
 
     # Create a raw image and mount it as a loop device
-    assert_script_run("qemu-img create -f raw $shared_dir/$img_filename.raw ${hdd_size}G");
-    my $device = script_output("losetup --find --show $shared_dir/$img_filename.raw");
+    assert_script_run(
+        "qemu-img create -f raw $shared_dir/$img_filename.raw ${hdd_size}G");
+    my $device =
+      script_output("losetup --find --show $shared_dir/$img_filename.raw");
```

Comment:
you could just use `qemu-nbd` to write into the qcow2 directly rather than doing a raw to qcow translation. see https://github.com/SUSE/elemental/blob/main/docs/building-linux-image.md#prepare-the-installation-target for details.